### ISSUE TITLE
MacOS startup time is so slow

### ISSUE THREAD

**Issue by vizh** [zed-industries/community#1745]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

The promotional materials on the website suggest that the startup time of Sublime and Zed should be practically the same. However, in reality, Sublime starts almost instantly, while Zed takes around 4-5 seconds to display its window. This refers to a clean installation without any additional settings.

### Environment

Zed: v0.93.4 (stable)
OS: macOS 13.4.1
Memory: 32 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

https://github.com/zed-industries/community/assets/167476/4201dbb9-e113-47d2-ba1e-16d091feda98

Recorded a video showing the difference in speed between running Zed and Sublime.

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.

[Zed.log](https://github.com/zed-industries/community/files/11997032/Zed.log)

**Comment by osiewicz** [comment]
Is the startup time slow with copilot disabled too? Based on the logs the request seems to be timing out.

**Comment by vizh** [comment]
> Is the startup time slow with copilot disabled too? Based on the logs you've provided the request seems to be timing out.

This is a clean installation. No configuration after being copied to the applications directory on a system where it was not previously installed. I checked just now, Copilot is turned off.

Disabling all network interfaces did not speed up the startup process. When no network interfaces are available, any connection attempt is instantly terminated without any waiting period.

**Comment by jansol** [comment]
Hmm, what buffer font have you set in your config? I managed to get very slow startup times with specific fonts in #1635

**Comment by vizh** [comment]
> Hmm, what buffer font have you set in your config? I managed to get very slow startup times with specific fonts in #1635

You're right! Just removed [Iosevka](https://github.com/be5invis/Iosevka) from config and problem was gone. This Issue is duplicate of #1635!

**Comment by iamnbutler** [comment]
> > Hmm, what buffer font have you set in your config? I managed to get very slow startup times with specific fonts in #1635
> 
> You're right! Just removed [Iosevka](https://github.com/be5invis/Iosevka) from config and problem was gone. This Issue is duplicate of #1635!

I've said this in other places, but Iosevka is MASSIVE. Honestly, outside of something that is preloaded like it is in Zed I really wouldn't recommend using it without subsetting the font to something smaller... It has hundreds of thousands of character glyphs in it by default, resulting in truly a massive font size (go compare the font size of a single weight of Iosevka to almost any other font, its 10x+ larger 😬)

**Comment by jansol** [comment]
> I've said this in other places, but Iosevka is MASSIVE.

I take it this refers specifically to it having way more control points per glyph than it needs rather than the number of glyphs? Since the problem does not occur with Arial Unicode MS (ships with macOS by default AFAIK).

**Comment by vizh** [comment]
It's interesting that I use this font everywhere, and I haven't noticed any software running slower with it. I can only assume that the font rendering or loading method using in Zed differs from the one used other software in MacOS. It would be odd to describe in the documentation for a programming editor which fonts are desirable or undesirable to use.

Thank you for the tip on trimming the font to only the used set of characters. I will give it a try as a temporary solution.

**Comment by vizh** [comment]
> Yeah, on our side we will need to see what is causing this. It's been an ongoing issue, but oddly has only showed up with this font.

If any debugging or other assistance is required from my end, I will be glad to help.

**Comment by mdogan** [comment]
I observe the similar slow startup times on macos with `Zed 0.98.2`. This is the same for a few recent versions too. Two subsequent startup times are:

```
mdogan:$ time zed

real	0m4.536s
user	0m0.021s
sys	0m0.017s
```

```
mdogan:$ time zed

real	0m4.531s
user	0m0.021s
sys	0m0.016s
```

User level `settings.json` is empty, nothing customized.

**Comment by JosephTLyons** [comment]
This should be fixed in [v0.119.9-pre](https://zed.dev/releases/preview/0.119.9) - please reopen if the issue persists.
